application/apps/indexer/sources/src/producer.rs

```diff
@@ -56,16 +56,19 @@ impl<T: LogMessage, P: Parser<T>, D: ByteSource> MessageProducer<T, P, D> {
         }
     }
     /// create a stream of pairs that contain the count of all consumed bytes and the
-    /// MessageStreamItem
-    pub fn as_stream(&mut self) -> impl Stream<Item = (usize, MessageStreamItem<T>)> + '_ {
+    /// MessageStreamItems in a boxed slice
+    pub fn as_stream(&mut self) -> impl Stream<Item = Box<[(usize, MessageStreamItem<T>)]>> + '_ {
         stream! {
-            while let Some(item) = self.read_next_segment().await {
-                yield item;
+            while let Some(items) = self.read_next_segment().await {
+                yield items;
             }
         }
     }
 
-    async fn read_next_segment(&mut self) -> Option<(usize, MessageStreamItem<T>)> {
+    // Ensure the compiler will inline this function as long as it's used once to improve compiler
+    // optimization.
```

Comment:
I would remove this inline directive. The function is pretty long and forcing the compiler to inline it might result in actually worse performance (cache misses, code bloat...)

Reply:
You are right. Originally, I've added this online to avoid calling `poll()` on more time on the function and inline it within the macros but this could affect other parts. 
It's better to leave it to the compiler to decide :+1: